src/util/arming.ts

```diff
@@ -0,0 +1,55 @@
+import path from "path";
+import fse from "fs-extra";
+
+const findupSync = require("findup-sync");
+
+function getDirIfIsFile(p: string): string {
+    return fse.existsSync(p) && fse.statSync(p).isFile() ? path.dirname(p) : p;
+}
+
+export function getBasePath(paths: string[]): string {
```

Comment:
This function will return "" if the user mixes absolute `/...` and relative `./...` paths. Perhaps it would be best if we normalize all paths to absolute in `detectProjectRoot` before we pass them to `getBasePath`?

Reply:
Addressed in https://github.com/ConsenSys/scribble/pull/133/commits/62dfc4eda8b6d526ae493944f6749a62b738b079